pkg/controller/atlasdatafederation/datafederation_controller.go

```diff
@@ -206,22 +222,11 @@ func (r *AtlasDataFederationReconciler) readProjectResource(ctx context.Context,
 }
 
 func (r *AtlasDataFederationReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	c, err := controller.New("AtlasDataFederation", mgr, controller.Options{Reconciler: r})
-	if err != nil {
-		return err
-	}
-
-	// Watch for changes to primary resource DataFederation & handle delete separately
-	err = c.Watch(&source.Kind{Type: &mdbv1.AtlasDataFederation{}}, &watch.EventHandlerWithDelete{Controller: r}, r.GlobalPredicates...)
-	if err != nil {
-		return err
-	}
-	err = c.Watch(&source.Kind{Type: &mdbv1.AtlasDataFederation{}}, &handler.EnqueueRequestForObject{}, r.GlobalPredicates...)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("AtlasDataFederation").
+		Watches(&source.Kind{Type: &mdbv1.AtlasDataFederation{}}, &watch.EventHandlerWithDelete{Controller: r}, builder.WithPredicates(r.GlobalPredicates...)).
```

Comment:
This looks very strange to me, but it does the same as before using the builder interface. We should visit this later.